device/platforms/stm32/HAL_Handlers.h

```diff
@@ -87,6 +92,14 @@ RetType register_spi_tx(SPI_HandleTypeDef* hspi, CallbackDevice* dev, int num);
 /// @return
 RetType register_spi_rx(SPI_HandleTypeDef* hspi, CallbackDevice* dev, int num);
 
+/// @brief register a device for GPIO complete callback
+/// @param halGPIO     the GPIO device to register the callback for
+/// @param dev      the device registering
+/// @param num      some unique number that will be passed back in the 'callback'
+///                 function of 'dev' when the event occurs
+/// @return
+RetType register_gpio(GPIO_HandleTypeDef* halGPIO, CallbackDevice* dev, int num);
```

Comment:
you don't actually need to callback number for simple GPIO, unless HAL differentiates between reads and writes

Reply:
but okay to leave if you want to match syntax and still use a dev_t